documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -2927,21 +2927,30 @@ In addition to controlling the execution mode using the `{Execution}` annotation
 Jupiter provides another annotation-based declarative synchronization mechanism. The
 `{ResourceLock}` annotation allows you to declare that a test class or method uses a
 specific shared resource that requires synchronized access to ensure reliable test
-execution. The shared resource is identified by a unique name which is a `String`. The
-name can be user-defined or one of the predefined constants in `{Resources}`:
+execution. The shared resource is identified by a unique name which is a `String`.
+The name can be user-defined or one of the predefined constants in `{Resources}`:
 `SYSTEM_PROPERTIES`, `SYSTEM_OUT`, `SYSTEM_ERR`, `LOCALE`, or `TIME_ZONE`.
 
-If the tests in the following example were run in parallel _without_ the use of
-{ResourceLock}, they would be _flaky_. Sometimes they would pass, and at other times they
-would fail due to the inherent race condition of writing and then reading the same JVM
-System Property.
+Since Junit Jupiter 5.12 `{ResourceLock}` annotation has 'providers' attribute
+which accepts an array of one or more classes implementing `{ResourceLocksProvider}`
+interface. This interface allows to add shared resources in runtime.
 
-When access to shared resources is declared using the `{ResourceLock}` annotation, the
-JUnit Jupiter engine uses this information to ensure that no conflicting tests are run in
-parallel. This guarantee extends to lifecycle methods of a test class or method. For
-example, if a test method is annotated with a `{ResourceLock}` annotation, the "lock" will
-be acquired before any `@BeforeEach` methods are executed and released after all
-`@AfterEach` methods have been executed.
+Note that resources declared "statically" with `{ResourceLock}` annotation
+are combined with resources added "dynamically" with `{ResourceLocksProvider}`
+interface.
+
+If the tests in the following examples were run in parallel _without_ the use of
+`{ResourceLock}` or `{ResourceLocksProvider}`, they would be _flaky_.
+Sometimes they would pass, and at other times they would fail due to
+the inherent race condition of writing and then reading the same JVM System Property.
+
+When access to shared resources is declared using the `{ResourceLock}` annotation
+or added with `{ResourceLocksProvider}` interface, the JUnit Jupiter engine uses
+this information to ensure that no conflicting tests are run in parallel.
+This guarantee extends to lifecycle methods of a test class or method.
+For example, if a test method is annotated with a `{ResourceLock}` annotation,
+the "lock" will be acquired before any `@BeforeEach` methods are executed
+and released after all `@AfterEach` methods have been executed.
```

Comment:
Please revert the changes made to the formatting of these two paragraphs.

Reply:
Done